addons/web_editor/static/src/js/common/ace-dialog.js

```diff
@@ -0,0 +1,109 @@
+/** @odoo-module **/
+
+import { loadBundle } from "@web/core/assets";
+import Dialog from 'web.Dialog';
+import core, { _t } from 'web.core';
+
+/**
+ * Render an code editor in the given node.
+ * @param {DOMElement} node where the editor will be rendered
+ * @param {String | undefined} content text of the editor
+ * @returns {Object} the ace editor instance
+ */
+export function renderCodeEditor(node, content) {
+    content = content || ''
+    const aceEditor = window.ace.edit(node);
+    aceEditor.setTheme('ace/theme/monokai');
+    aceEditor.setValue(content, 1);
+    aceEditor.setOptions({
+        minLines: 20,
+        maxLines: Infinity,
+        showPrintMargin: false,
+    });
+    aceEditor.renderer.setOptions({
+        highlightGutterLine: true,
+        showInvisibles: true,
+        fontSize: 14,
+    });
+
+    const aceSession = aceEditor.getSession();
+    aceSession.setOptions({
+        mode: "ace/mode/qweb",
+        useWorker: false,
+    });
+    return aceEditor;
+}
+
+/**
+ * Display a dialog to edit the custom code of the website.
+ * @param {OdooWidget} ctx
+ */
+export async function openHeadBodyEditor(widget) {
+    await loadBundle({
+        jsLibs: [
+            '/web/static/lib/ace/ace.js',
+            '/web/static/lib/ace/mode-xml.js',
+            '/web/static/lib/ace/mode-qweb.js',
+        ],
+    });
+
+    let websiteId;
+    widget.trigger_up('context_get', {
+        callback: (ctx) => {
+            websiteId = ctx['website_id'];
+        },
+    });
+
+    let website = (await widget._rpc({
+        model: 'website',
+        method: 'read',
+        args: [[websiteId], ['custom_code_head', 'custom_code_footer']],
+    }))[0]
+
+    const $content = $(core.qweb.render('web_editor.custom_code_dialog_content', {
+        headContentText: _t('Enter code that will be added into the <head> of every page of your site.'),
+        bodyContentText: _t('Enter code that will be added before the </body> of every page of your site.'),
```

Comment:
Why are those not part of the qweb template?

Reply:
It wasn't in the old code, but yeah I should move them there